src/main/java/budgetbuddy/logic/parser/commandparsers/loancommandparsers/MultiLoanCommandParser.java

```diff
@@ -0,0 +1,73 @@
+package budgetbuddy.logic.parser.commandparsers.loancommandparsers;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import budgetbuddy.commons.core.index.Index;
+import budgetbuddy.logic.commands.loancommands.MultiLoanCommand;
+import budgetbuddy.logic.parser.CommandParser;
+import budgetbuddy.logic.parser.CommandParserUtil;
+import budgetbuddy.logic.parser.exceptions.ParseException;
+import budgetbuddy.model.loan.util.PersonLoanIndexPair;
+
+/**
+ * Represents a command parser that can handle multiple loan targets.
+ */
+public abstract class MultiLoanCommandParser implements CommandParser<MultiLoanCommand> {
+
+    protected List<PersonLoanIndexPair> personLoanIndexPairs;
+    protected List<Index> personIndices;
+
+    public MultiLoanCommandParser() {
+        personLoanIndexPairs = new ArrayList<PersonLoanIndexPair>();
+        personIndices = new ArrayList<Index>();
+    }
+
+    /**
+     * Parses multiple loan targets into two lists to be passed on to Command classes.
+     * {@code personLoanIndexPairs} is used for loan targets that a subset of a person's loans.
```

Comment:
Sorry, there's a typo in the Javadoc, should be `targets that are a subset` . A user can target multiple loans by typing something like `loan paid 1.1 2.1-3-4 3` . This list is used to handle `1.1` and `2.1-3-4` , which target person 1's loan 1 only (instead of all of person 1's loans) and person 2's loan 1, 3, and 4 only respectively.

Reply:
Got it, looks good!